src/instructlab/config/init.py

```diff
@@ -132,12 +132,14 @@ def init(
             )
             raise click.exceptions.Exit(1)
 
-    cfg.chat.model = model_path
-    cfg.generate.model = model_path
-    cfg.serve.model_path = model_path
+    # we should not override all paths with the serve model if special ENV vars exist
+    if param_source != click.core.ParameterSource.ENVIRONMENT:
+        cfg.chat.model = model_path
+        cfg.generate.model = model_path
+        cfg.serve.model_path = model_path
+        cfg.evaluate.model = model_path
```

Comment:
I feel like this all needs to be revisited after GA - seems like a band aid solution

Reply:
yes, I agree

Reply:
Indeed. I think the culprit is the fact that the CLI argument has an "acceptable" default that overrides regardless of what is in the config. The default value for `model-path` should be made `None`, in which case we should not touch what's pulled from the config (either env or default).

Reply:
(Something like [this](https://github.com/instructlab/instructlab/pull/2014).)